src/build_log.h

```diff
@@ -15,6 +15,16 @@
 #ifndef NINJA_BUILD_LOG_H_
 #define NINJA_BUILD_LOG_H_
 
+#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900)
+#define NINJA_NULLPTR nullptr
+#else
+#define NINJA_NULLPTR NULL
+#endif
+
+#if __cplusplus >= 201103L || (defined(_MSC_VER) && _MSC_VER >= 1900)
+#include <memory>
```

Comment:
Why bother protecting this include? If nothing from it is used, it's a no-op. Just #include <memory> without the ifdefs.